pycode/memilio-epidata/memilio/epidata/getCaseData.py

```diff
@@ -0,0 +1,676 @@
+    df.loc[df.Altersgruppe == 'unbekannt', [
+        'Altersgruppe']] = dd.GerEng['unbekannt']
+
+    # change names of columns
+    df.rename(dd.GerEng, axis=1, inplace=True)
+
+    # Add column 'Date' with Date
+    # = reporting date if rep_date is set
+    # = reference date (date of disease onset) if IstErkrankungsbeginn = 1 else
+    #       take Meldedatum (reporting date)
+    if rep_date:
+        df['Date'] = df['Meldedatum'].astype('object')
+    else:
+        df['Date'] = np.where(
+            df['IstErkrankungsbeginn'] == 1, df['Refdatum'],
+            df['Meldedatum'])
+
+    # Correct Timestamps:
+    for col in ['Date']:
+        df[col] = df[col].astype('datetime64[ns]')
+
+    # Date is either Refdatum or Meldedatum after column
+    # 'IstErkrankungsbeginn' has been added. See also rep_date option.
+    dateToUse = 'Date'
+    df.sort_values(dateToUse, inplace=True)
+
+    # Manipulate data to get rid of conditions: df.NeuerFall >= 0, df.NeuerTodesfall >= 0, df.NeuGenesen >=0
+    # There might be a better way
+    df.loc[df.NeuerFall < 0, [AnzahlFall]] = 0
+    df.loc[df.NeuerTodesfall < 0, [AnzahlTodesfall]] = 0
+    df.loc[df.NeuGenesen < 0, [AnzahlGenesen]] = 0
+
+    # get rid of unnecessary columns
+    df = df.drop(['NeuerFall', 'NeuerTodesfall', 'NeuGenesen',
+                 "IstErkrankungsbeginn", "Meldedatum", "Refdatum"], 1)
+
+    print("Available columns:", df.columns)
+
+    ######## Data for whole Germany all ages ##########
+
+    # NeuerFall: Infected (incl. recovered) over "dateToUse":
+
+    # make sum for one "dateToUse"
+    gbNF = df.groupby(dateToUse).agg({AnzahlFall: sum})
+
+    # make cumulative sum of "AnzahlFall" for "dateToUse"
+    gbNF_cs = gbNF.cumsum()
+
+    # output to json file
+    # prefix for all files about cases data (apart from raw data)
+    prefix = "cases_"
+    filename = 'infected'
+    if rep_date:
+        filename_orig = filename + '_repdate'
+    else:
+        filename_orig = filename
+    gd.write_dataframe(gbNF_cs.reset_index(), directory,
+                       prefix + filename_orig, file_format)
+    if impute_dates or moving_average > 0:
+        gbNF_cs = modifyDataframeSeries.impute_and_reduce_df(
+            gbNF_cs.reset_index(),
+            {},
+            ['Confirmed'],
+            impute='forward', moving_average=moving_average)
+        filename = gd.append_filename(filename, impute_dates, moving_average)
+        if rep_date:
+            filename = filename + '_repdate'
+        gd.write_dataframe(gbNF_cs, directory, prefix + filename, file_format)
+
+    if make_plot:
+        # make plot
+        gbNF_cs.plot(title='COVID-19 infections', grid=True,
+                     style='-o')
+        plt.tight_layout()
+        plt.show()
+
+    # Dead over Date:
+    gbNT = df.groupby(dateToUse).agg({AnzahlTodesfall: sum})
+    gbNT_cs = gbNT.cumsum()
+    gbNT_cs = gbNT_cs[gbNT_cs[AnzahlTodesfall] != 0]
```

Comment:
I'm not sure anymore why I added line 285. There are some dates in the beginning where the number of deaths is still zero. Those are removed by this line but I think this is not necessary. I would delete this line entirely. 

The condition `df.NeuerTodesfall >= 0` is not necessary anymore. Since the dataframe `df` only appeared with this condition, I changed line 235 to change the dataframe beforhand.

Reply:
Having rows with zeros in the beginning makes a difference when computing the moving average if there are zero lines in the beginning. I added line 285 to not change the behaviour in this case. I can move it before computing the cumulative sum.